megatron/data/gpt_dataset.py

```diff
@@ -48,27 +50,63 @@ def build_train_valid_test_datasets(data_prefix, data_impl, splits_string,
 
     # Build individual datasets.
     train_datasets = []
+    # we'll temporarily store the validation sets then compare them with the arguments next step
+    # this needs to be ordered so it lines up with the weights
+    ds_shared_with_train = OrderedDict()
     valid_datasets = []
     test_datasets = []
-    for i in range(len(prefixes)):
+    for i, prefix in enumerate(prefixes):
         train_ds, valid_ds, test_ds = _build_train_valid_test_datasets(
-            prefixes[i], data_impl, splits_string,
+            prefix, data_impl, splits_string,
             datasets_train_valid_test_num_samples[i],
             seq_length, seed, skip_warmup)
+        print_rank_0(f"split: {splits_string}")
         if train_ds:
             train_datasets.append(train_ds)
+            print_rank_0(f"train_ds size: {len(train_ds)}")
         if valid_ds:
-            valid_datasets.append(valid_ds)
+            # a safe split without overlap with training that we may use later if `valid_data_prefix` includes it
+            ds_shared_with_train[prefix] = valid_ds
+            print_rank_0(f"valid_ds size: {len(valid_ds)}")
         if test_ds:
             test_datasets.append(test_ds)
 
+    if valid_data_prefix is not None:
+        # in this case `valid_data_prefix` defines what is in the validation mix
+        # we make sure there is no overlap with the training mix by comparing with ds_shared_with_train
+        valid_output = get_datasets_weights_and_num_samples(valid_data_prefix,
+                                                  [0, train_valid_test_num_samples[1], 0])
+        valid_prefixes, valid_weights, valid_datasets_samples = valid_output
+        for i, prefix in enumerate(valid_prefixes):
+            if prefix not in ds_shared_with_train:
+                print_rank_0(f"prefix: {prefix} not found in {ds_shared_with_train.keys()}")
```

Comment:
```suggestion
                print_rank_0(f"prefix: {prefix} not found in {ds_shared_with_train.keys()}")
```
This looks more like a debug and won't be of any added value to the user, IMHO, and just add to the noise.

What do you think?